### PR TITLE
Binary datasets and training code rewrite

### PR DESCRIPTION
**Pull request by mittagessen**
This branch contains a complete overhaul of the training code using pytorch lightning and a new binary text recognition dataset format based on the Arrow IPC format.

- [X] Binary dataset builder for baseline data
- [X] Binary dataset builder for bbox data
- [x] Training of bbox/baseline data in new binary format
- [x] Line-level fixed splits
- [X] Learning rate schedulers
- [x] Documentation
- [x] Tests

This will effectively address #276.

### CODE REVIEW

**Comment by mittagessen** [comment]
github doesn't seem to pick up the manual squash merge.
